### ISSUE TITLE
Buit-in webserver injects path segments when accessing correct rendered {% url %}

### ISSUE THREAD

**Issue by tombreit** [thanethomson/statik#49]
I get unexpected urls when accessing the site through the built-in webserver, but the html is rendered correct: 

 in | out  
------------ | -------------
**template** | ``{% url 'album' album %}``
**html** |  ``/projects/json2statik/album/eguelshardt-elsass``
**built-in webserver** | ``http://localhost:8000/projects/json2tt/album/album/eguelshardt-elsass/``

The ``/projects/json2statik`` is defined as ``base-path``. 

On the production site it's just fine: the url ``/projects/json2statik/album/eguelshardt-elsass`` took me to the expected ressource ``/projects/json2statik/album/eguelshardt-elsass`` - but with the built-in webserver I get redirected to ``http://localhost:8000/projects/json2statik/album/album/eguelshardt-elsass/`` - where does this duplication of ``album`` came from? Any tips or hints?

## Involved  views

I link from ``album_tile_index.yml``:
```
# views/album_tile_index.yml
path: /albums/browse
template: album_tile_index
context:
  dynamic:
    albums: session.query(Album).order_by(Album.sort_by.desc())
```
to ``album.yml``:
```
# views/album.yml
path:
  template: /album/{{ album.pk }}
  for-each:
    album: session.query(Album).all()
template: album
```

**Comment by tombreit** [comment]
... since statik does generate the correct html, it may actually a bug in httpwatcher (https://github.com/thanethomson/httpwatcher)?

**Comment by dcferreira** [comment]
I also have this problem, but it only happens with statik version 0.18.1 (it works with both 0.17.0 and 0.18.0), so I guess it's something in statik itself.

**Comment by thanethomson** [comment]
So, after much investigation, it turned out it was an issue in `httpwatcher` after all 😁 Basically, when navigating to a URL without a trailing forward-slash, using a non-standard base path, while using the embedded web server, `httpwatcher` would incorrectly redirect the request.

It's now been fixed in `httpwatcher` v0.5.1, and I've updated Statik in v0.18.2 to depend on `httpwatcher` v0.5.1.

If you're still getting issues with the incorrect redirect, your browser's cache may have to be cleared. I had to clear my cache in Firefox to get it to work properly again, even after installing the new version of Statik.
